src/components/utility/Button/Button.jsx

```diff
@@ -1,31 +1,54 @@
 import React from "react";
-import styles from './Button.module.css';
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import styles from "./Button.module.css";
 
-const Button = ({ type = "primary", disabled = false, onClick, children }) => {
-  let buttonClass = styles.button;
+const buttonColorStyle = {
+  white: {
+    color: "#fff",
+    borderColor: "#fff",
+    hoverBackgroundColor: "white",
+    hoverColor: "#002529",
+  },
+  dark: {
+    color: "#012F34",
+    borderColor: "#012F34",
+    hoverBackgroundColor: "#012F34",
+    hoverColor: "#fff",
+  },
+};
+
+const Button = ({ style, disabled = false, onClick, children, className }) => {
+  const buttonClass = classNames(styles.button, className, {
+    [styles.disabled]: disabled,
+  });
 
-  switch (type) {
-    case 'primary':
-      buttonClass += ` ${styles.buttonPrimary}`;
-      buttonClass += !disabled ? ` ${styles.buttonPrimaryHover}` : ` ${styles.buttonPrimaryDisabled}`;
-      break;
-    case 'secondary':
-      buttonClass += ` ${styles.buttonSecondary}`;
-      buttonClass += !disabled ? ` ${styles.buttonSecondaryHover}` : ` ${styles.buttonSecondaryDisabled}`;
-      break;
-    case 'tertiary':
-      buttonClass += ` ${styles.buttonTertiary}`;
-      buttonClass += !disabled ? ` ${styles.buttonTertiaryHover}` : ` ${styles.buttonTertiaryDisabled}`;
-      break;
-    default:
-      buttonClass += ` ${styles.buttonPrimary}`;
-      buttonClass += !disabled ? ` ${styles.buttonPrimaryHover}` : ` ${styles.buttonPrimaryDisabled}`;
-  }
+  const colorConfig = {
+    "--button-color": buttonColorStyle[style].color,
+    "--button-border-color": buttonColorStyle[style].borderColor,
```

Comment:
pros/cons of adding a local var like `currentStyle = buttonColorStyle[style]`? Would keep lines shorter, might be an improvement to readability, but adds a variable where we don't strictly need one